samza-core/src/main/java/org/apache/samza/runtime/LocalApplicationRunner.java

```diff
@@ -200,6 +283,21 @@ static String createProcessorId(ApplicationConfig appConfig) {
     }
   }
 
+  private void cleanup() {
+    if (runIdGenerator.isPresent()) {
+      runIdGenerator.get().close();
```

Comment:
nit: can be simplified with `runIdGenerator::ifPresent(RunIdGenerator::close);`
same for `coordinationUtils`